.github/workflows/mega-linter.yml

```diff
@@ -0,0 +1,182 @@
+# MegaLinter GitHub Action configuration file
+# More info at https://megalinter.io
+---
+name: MegaLinter
+
+# Trigger mega-linter at every push. Action will also be visible from Pull
+# Requests to main
+on: # yamllint disable-line rule:truthy - false positive
+  # Comment this line to trigger action only on pull-requests
+  # (not recommended if you don't pay for GH Actions)
+  push:
+
+  pull_request:
+    branches:
+      - main
+      - master
+
+# # Comment env block if you do not want to apply fixes
+# env:
+#   # Apply linter fixes configuration
+#   #
+#   # When active, APPLY_FIXES must also be defined as environment variable
+#   # (in github/workflows/mega-linter.yml or other CI tool)
+#   APPLY_FIXES: all
+
+#   # Decide which event triggers application of fixes in a commit or a PR
+#   # (pull_request, push, all)
+#   APPLY_FIXES_EVENT: pull_request
+
+#   # If APPLY_FIXES is used, defines if the fixes are directly committed (commit)
+#   # or posted in a PR (pull_request)
+#   APPLY_FIXES_MODE: commit
+
+concurrency:
+  group: ${{ github.ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+permissions:  # added using https://github.com/step-security/secure-repo
+  contents: read
+
+jobs:
+  megalinter:
+    name: MegaLinter
+    runs-on: ubuntu-latest
+
+    # Give the default GITHUB_TOKEN write permission to commit and push, comment
+    # issues & post new PR; remove the ones you do not need
+    permissions:
+      contents: write
+      issues: write
+      pull-requests: write
+
+    steps:
+
+      - name: Harden Runner
+        uses: step-security/harden-runner@91182cccc01eb5e619899d80e4e971d6181294a7 # v2.10.1
+        with:
+          egress-policy: audit
```

Comment:
I would move this one out of megalinter

Is the file a copy paste of a kind of megalinter template?

Reply:
> I would move this one out of megalinter
Okay, I can do that. I was trying to follow the other actions. 

> Is the file a copy paste of a kind of megalinter template?
Yes. It is. It serves as a good starting point. I have removed some configurations, for example we do not need the linter to make fixes. I have also addded some, for example to diable checking the test fixtures.

Reply:
Leave harden there then, let's wait a maintainer feedback

Reply:
OpenSSF recommends using the harden-runner so would like to keep it for consistency